process/block/baseProcess.go

```diff
@@ -381,7 +381,8 @@ func (bp *baseProcessor) requestHeadersIfMissing(sortedHdrs []data.HeaderHandler
 		return err
 	}
 
-	if len(sortedHdrs) == 0 {
+	isLastNotarizedCloseToOurRound := maxRound-prevHdr.GetRound() <= process.MaxHeaderRequestsAllowed
```

Comment:
maybe isLastNotarizedClose**d**ToOurRound ?

Reply:
no. it is close, gen aproape :)